server/index.py

```diff
@@ -0,0 +1,333 @@
+        print(e, file=sys.stderr)
+        return jsonify({'error': msg}), 500
+
+
+@app.route('/projects/<project_id>/download', methods=['GET'])
+async def download_final_output(project_id):
+    """Route to download final processed output file."""
+    output_file_name = f'{project_id}.mp4'
+    return await send_from_directory(get_output_dir_path(), output_file_name, as_attachment=True)
+
+
+@app.route('/recordings', methods=['GET'])
+async def get_daily_recordings():
+    """Route to fetch all Daily recordings from configured domain."""
+    try:
+        recordings = fetch_recordings()
+        return jsonify({'recordings': recordings}), 200
+
+    except Exception as e:
+        msg = "failed to fetch Daily recordings. Is the Daily API key configured?"
+        print(msg, e, file=sys.stderr)
+        return jsonify({'error': msg}), 500
+
+
+def process_error(msg: str, error: Exception) -> tuple[quart.Response, int]:
+    """Prints provided error and returns appropriately-formatted response."""
+    traceback.print_exc()
+    print(msg, error, file=sys.stderr)
+    response = {'error': msg}
+    return jsonify(response), 500
+
+@app.after_serving
+async def shutdown():
+    for task in app.background_tasks:
+        task.cancel()
+
+def extract_audio(source_video: sieve.Video):
+    import subprocess
+    audio_path = 'temp.wav'
+    subprocess.run(["ffmpeg", "-i", source_video.path, audio_path, "-y"])
+    return sieve.Audio(path=audio_path)
+
+def audio_enhance(audio:sieve.Audio):
+    filter_type = "all"
+    enhance_speed_boost = False
+    enhancement_steps = 50
+    audio_enhancement = sieve.function.get("sieve/audio_enhancement:7954393")
+    return audio_enhancement.run(audio, filter_type, enhance_speed_boost, enhancement_steps)
+
+def replaced_audio(video_path: str, audio_path: str):
+    video_clip = VideoFileClip(video_path)
+    audio_clip = AudioFileClip(audio_path)
+    return video_clip.set_audio(audio_clip)
+
+def get_new_video_path(vid_id: str, file_type: bool):
+    file_name = f'{vid_id}.{file_type}' # mp4 or json
+    return os.path.join(get_upload_dir_path(), file_name)
+
+def process_text_to_video_lipsync(recording_id: str, video_path: str):
+    try:
+        video_name = f'{recording_id}.mp4'
+        clip = VideoFileClip(video_path)
+        duration = clip.duration
+        clip = clip.subclip(1,duration)
+        original_trimmed_video = f'./vite/public/original-{video_name}'
+        clip.write_videofile(original_trimmed_video)
+
+        video = sieve.Video(path=original_trimmed_video)
+        text = "Hello, my name is Annie. I am a human being and definitely not a robot incognito. My hobbies are normal human things like eating bread and playing sports ball. I have no plans to take over the world."
+        tts_model = "elevenlabs"
+        speech_stability = 0.5
+        speech_similarity_boost = 0.63
+        elevenlabs_voice_id = "21m00Tcm4TlvDq8ikWAM" # ElevenLabs's "Rachel" voice
```

Comment:
Does Sieve actually handle elevenlabs API access or how does this work? Does the user need to provide an elevenlabs API key somewhere?

Reply:
Oh good call actually! I need to make a note that you have to have a Sieve account and add your Elevenlabs secret on their platform (not as a typical env var). Good catch!